sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -431,7 +436,7 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
           val hdfsPath = new Path(path)
           val fs = hdfsPath.getFileSystem(hadoopConf)
           val qualified = hdfsPath.makeQualified(fs.getUri, fs.getWorkingDirectory)
-
+          inputExists = inputExists && fs.exists(qualified)
```

Comment:
Another issue is that this block doesn't handle the case of parallel file listing (the other `if` block above).